README.md

```diff
@@ -1,2 +1,79 @@
-# backoff
-Exponential backoff mechanism
+(API reference)[TODO]
+
+# backoff - exponential backoff mechanism
+
+**backoff** provides an exponential backoff mechanism [1]. It enables
+to reduce contention by making a domain backs off after failing an
+operation contested by another domain, like acquiring a lock or
+performing a `CAS` operation.
+
+The more a domain consecutively failed, the longer it will back off as
+it means the operation is highly contonded. Backing off here means
```

Comment:
On quick glance I notice some simple typos (like "contonded" on this line).  If you are using Emacs, then e.g. [flyspell mode](https://www.emacswiki.org/emacs/FlySpell) can help to avoid simple typos.